.github/workflows/auto-reset-parent-or-child-lab-machine.yml

```diff
@@ -79,12 +81,16 @@ jobs:
     - name: Start VM, wait for online status, alert observer.
       run: |
         $vmName = "${{ matrix.vec.vm-name }}"
-        Start-VM -Name $vmName
-        while (-not (Get-VMNetworkAdapter -VMName $vmName).IPAddresses) {
-          Write-Host "Waiting for VM to be online..."
-          Start-Sleep -Seconds 5
+        if ($vmName.Contains("linux")) {
```

Comment:
It's not great that we have to repeat this code 4 different times.

Reply:
I agree, I could make a follow up PR and/or issue to merge some of the behavior into 1 big workflow.

My concern mainly is understandability of the code. If we were to merge all 4 scenarios, we would have a bunch of conditionals and logic that may make it difficult to interpret for maintainers. 

The business logic here enables these workflows to be triggered as part of a manual dispatch (to fix something obviously broken), a scheduled dispatch, and an automatic dispatch. Each scenario requires something a little bit different.

Reply:
Collapsed the 2 workflows into 1.